File: client/src/ts/brush.ts
import { BrushData, Encodable } from "./types";
import { abconcat, str2ab } from "./utils";

export class Brush implements Brush, Encodable {
  public lineWidth = 3;
  public strokeStyle = "#333333";

  public encode() {
    const strokeStyle = str2ab(this.strokeStyle);
    const lineWidth = Uint16Array.from([this.lineWidth]).buffer;

    const buff = abconcat(strokeStyle, lineWidth);

    return buff;
  }
}

File: server/src/paper/brush.rs
#[derive(Debug)]
pub(crate) struct Brush {
    pub stroke_style: String,
    pub line_width: i32,
}

impl Brush {
    pub fn new() -> Brush {
        Brush {
            stroke_style: String::from("#333333"),
            line_width: 3,
        }
    }
}
